src/environments/array.js

```diff
@@ -85,8 +88,8 @@ function parseArray(
         }
     }
 
-    let row = [];
-    const body = [row];
+    let lastRow = [];
```

Comment:
Why the name change?

Reply:
I removed the local variable `lastRow` below as it is always the same as this one (whose type the inference can easily figure out). I renamed this to `lastRow` since it's relevant in that context to make the semantics more obviously correct. I can renamed it back if you prefer.

Reply:
Reusing this variable in the `else if (next === "\\end")` block makes sense.  Since there can be multiple rows the variable doesn't hold strictly the last row I think that calling it `row` would be more accurate.

Reply:
Doesn't it always hold the last row? Below, whenever it's assigned a new value, it's also pushed to the end of `body`. I don't see any other pushes into `body`.

Reply:
Inside the `while` loop it's gets push to `body` whenever there's a `\\cr` and a new one is created at the same time.

Reply:
Ah, I think I understand our difference in interpretations. Just to be on the same page: my interpretation of
```
lastRow = [];
body.push(lastRow);
```
is that at after these two lines, `lastRow` still points to what's *currently* the last row in `body` (as opposed to final last row). It made its role in the rest of the logic a bit clearer to me, but I can understand the confusion due to a different interpretation.
Renamed back to `row`.